Formula/ts-node.rb

```diff
@@ -0,0 +1,40 @@
+require "language/node"
+require "json"
+
+class TsNode < Formula
+  desc "TypeScript execution and REPL for node.js"
+  homepage "https://github.com/TypeStrong/ts-node"
+  url "https://registry.npmjs.org/ts-node/ts-node-8.0.2.tgz"
+  sha256 "ba6a690ddf1e441425fe47d37b06e61fe4a8d3be10430b17eb286630d9569324"
+
+  depends_on "node"
+
+  resource "typescript" do
+    url "https://registry.npmjs.org/typescript/-/typescript-3.3.1.tgz"
+    sha256 "5693094bc766af02ec381114cb933c1f6ffc78c1395a4c42b55b3ccf48aa4f50"
+  end
+
+  def install
+    (buildpath/"node_modules/typescript").install resource("typescript")
+
+    # declare typescript as a bundledDependency of ts-node
+    pkg_json = JSON.parse(IO.read("package.json"))
+    pkg_json["dependencies"]["typescript"] = resource("typescript").version
+    pkg_json["bundledDependencies"] = ["typescript"]
+    IO.write("package.json", JSON.pretty_generate(pkg_json))
+
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
```

Comment:
That's unfortunate that `ts-node` expects to be requireable `typescript` in the current working directory (and sets the require `cwd` options that way. And obvious workaround would be to use a `env_script` and set nodes `NODE_PATH` explicitly to the requireable location of `typescript`. But this is only a dirty way working around the
fact that the `ts-node` developer doesn't like global installs.
```suggestion
    (bin/"ts-node").write_env_script libexec/"bin/ts-node", :NODE_PATH => "#{libexec}/lib/node_modules/ts-node/node_modules/"
```

Reply:
Thank you and that would be very kind for the detailed suggestion! I'll leave the commit for a moment until the maintainer of `ts-node` respond with any cause and fix. It seems that `ts-node` would work globally by running `$ npm install -g ts-node typescript` and I haven't figure out why.

Reply:
It works because `/usr/local/lib/node_modules` (where npm installs stuff globally) is in nodes global search path. Even after their local search for typescript fails (because they’re making it relative to the cwd) it’s still found by node by looking in the global path. That’s how our workaround function by adding an additional global search path and thta‘s why global default npm installations are working but global yarn installations are not (yarn is using a custom global module folder in the users home directory, which you have to add to your path when using yarn installed global modules).

Reply:
Thanks! I didn't quite understand well about how `require()` works. You're a Node guru to me and I learned a lot.

Reply:
I'm just contributing to the `node` related part of homebrew for quite some time now.

I think the upstream motivation for making the `typescript` require relative to the `cwd` was so that you are able to use a globally installed `ts-node` with any locally installed `typescript` compiler in your project (and more importantly that the locally installed version has preference over a globally installed one, with a potential incompatible version). Unfortunately this breaks some full global setups (yarn, homebrew ...). But upstream seems to recommend to install both `ts-node` and the `typescript` compiler locally and because of that I think it's quite unlikely that they will revert this change.

Reply:
Thanks! I believe this is the reasonable notion behind the local/ global dependencies design.